packages/cli/src/util/deploy/process-deployment.ts

```diff
@@ -281,3 +285,23 @@ export default async function processDeployment({
     throw err;
   }
 }
+
+export const archiveSuggestionText =
+  'Try using `--archive=tgz` to limit the amount of files you upload.\nLearn more: https://vercel.com/docs/cli/deploy#archive';
+
+export function handleErrorSolvableWithArchive(error: unknown) {
+  const errorIsObject = typeof error === 'object' && error !== null;
+  if (errorIsObject && 'message' in error) {
+    const isUploadRateLimit =
+      'code' in error &&
+      error.code === 'rate_limited' &&
+      typeof error.message === 'string' &&
+      error.message.includes('api-upload');
```

Comment:
Can we match on something stronger than a portion of a string error message for this?

Reply:
Fixed! Turns out our client code was removing the specific rate limit name field higher up in the levels of try catch: https://github.com/vercel/vercel/blob/697160902eb68ba02a8bd2b24c0b42836600611c/packages/client/src/upload.ts#L179. Edited `DeploymentError` to include `rateLimitName` and used that.